File: C++/Number/program21.cpp
//Accept the number from user and display the addition of that number from 1 to that number.

// Input : 4
// Output : 10  (1+2+3+4)

#include <iostream>
using namespace std;

class Number
{
    public:
        int iNo;

        void Accept()           // Setter
        {
            cout << "Enter the number" << endl;
            cin >> iNo;
        }

        void Display()          // Getter
        {
            cout<<"The numbers is: "<< this->iNo <<endl;
        }

        int Addition()
        {
            int iCnt = 0;
            int iSum = 0;

            if(iNo < 0)
            {
                iNo = -iNo;
            }
            
            for(iCnt = 1; iCnt <= iNo; iCnt++)
            {
                iSum = iSum + iCnt;
            }
            return iSum;
        }
};

int main()
{
    Number nobj;

    int iRet = 0;

    nobj.Accept();
    nobj.Display();

    iRet = nobj.Addition();
    cout << "Addition is: " << iRet << endl;

    return 0;
}

File: Java/Digit/program21.java
//Accept two numbers from user and display the power

// Input : 2    4
// Output : 16  (2 * 2 * 2 * 2)

import java.util.*;

class Digit
{
    public int iNo1;
    public int iNo2;

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the base");
        iNo1 = sobj.nextInt();

        System.out.println("Enter the power");
        iNo2 = sobj.nextInt();
    }

    public void Display()
    {
        System.out.println("The base is: "+this.iNo1);
        System.out.println("The power is: "+this.iNo2);
    }

    public long Power()
    {
        int iCnt = 0;
        long lMult = 1;

        for(iCnt = 1; iCnt <= iNo2; iCnt++)
        {
            lMult = lMult * iNo1;
        }
        return lMult;
    }
    
}

class program21
{
    public static void main(String arg[])
    {
        long lRet;

        Digit obj = new Digit();

        obj.Accept();
        obj.Display();

        lRet = obj.Power();
        System.out.println("Result is : "+lRet);
    }
}